pkg/target/host.go

```diff
@@ -158,9 +158,14 @@ func (h *Host) Flush(d time.Duration) {
 		case <-t.C:
 			h.Wm.Lock()
 			if h.W != nil {
-				err := h.W.Flush()
-				if err != nil {
-					h.Lg.Error("error while flushing the host buffer", zap.Error(err), zap.String("host name", h.Name), zap.Uint16("host port", h.Port))
+				h.Lg.Info(fmt.Sprintf("buffered %d", h.W.Buffered()))
+				if h.W.Buffered() != 0 {
+					err := h.W.Flush()
+					if err != nil {
+						h.Lg.Error("error while flushing the host buffer", zap.Error(err), zap.String("host name", h.Name), zap.Uint16("host port", h.Port))
+						// if flushing fails, the connection has to be re-established
+						h.Connect()
```

Comment:
This can cause:
 a) deadlock on h.Wm.Lock(). We are acquiring this lock in this function(line 159) also inside Connect().
 b) race conditions on accessing h.Conn. Not sure what if Dial() is safe or not.

Reply:
:+1:

Reply:
Fixed that, please check again.